File: prev_exams_and_ans/2015ccc/q3_rovarspraket/Rovarspraket.java
import java.io.IOException;
import java.util.Scanner;

public class Rovarspraket {

	private static String alphabet = "abcdefghijklmnopqrstuvwxyz";
	public static void main(String[] args) throws IOException {
		Scanner sc = new Scanner(System.in);
		String word = sc.nextLine();
		sc.close();
		
		String outputString = "";
		for (int i = 0; i < word.length(); i++) {
			char c = word.charAt(i);
			if (isVowel(c)) {
				outputString += c;
			} else /* isConsonant */ {
				outputString += transform(c); 
			}
		}
		System.out.println(outputString);
		
//		System.out.println(nearestVowel('j'));
	}
	
	public static boolean isVowel(char c) {
		return "aeiou".indexOf(c) >= 0;
	}
	
	public static String transform(char c) {
		return "" + c + nearestVowel(c) + nextConsonant(c);
	}
	
	public static char nextConsonant(char c) {
		int indexOfC = alphabet.indexOf(c);
		for (int i = indexOfC + 1; i < alphabet.length(); i++) {
			char nextChar = alphabet.charAt(i);
			if (!isVowel(nextChar))
				return nextChar;
		}
		return 'z';
	}
	
	public static char nearestVowel(char c) {
		int idxOfC = alphabet.indexOf(c);
		int idxOfNextVowel = 100;
		int idxOfPrevVowel = -100;
		for (int i = idxOfC - 1; i >= 0; i--) {
			if (isVowel(alphabet.charAt(i))) {
				idxOfPrevVowel = i;
				break; // can be tested on class
			}
		}
		
		for (int i = idxOfC + 1; i < alphabet.length(); i++) {
			if (isVowel(alphabet.charAt(i))) {
				idxOfNextVowel = i;
				break; // can be tested on class
			}
		}
		
		return (idxOfC - idxOfPrevVowel <= idxOfNextVowel - idxOfC) 
				? alphabet.charAt(idxOfPrevVowel) 
				: alphabet.charAt(idxOfNextVowel);
	}
}

File: prev_exams_and_ans/2015ccc/q3_rovarspraket/rovarspraket.py
alphabet = "abcdefghijklmnopqrstuvwxyz"

word = input()

def is_vowel(c):
    return "aeiou".find(c) >= 0

def transform(c):
    return c + nearest_vowel(c) + next_consonant(c)


def nearest_vowel(c):
    index_of_c = alphabet.find(c)
    index_of_next_vowel = 100
    index_of_prev_vowel = -100
    for i in range(index_of_c-1, -1, -1):
        # starting from index_of_c - 1, decrease by 1, and large than -1(exclude -1)
        if (is_vowel(alphabet[i])):
            index_of_prev_vowel = i
            break

    for i in range(index_of_c+1, len(alphabet)):
        if (is_vowel(alphabet[i])):
            index_of_next_vowel = i
            break
    
    return (alphabet[index_of_prev_vowel] 
            if index_of_c - index_of_prev_vowel <= index_of_next_vowel - index_of_c 
            else alphabet[index_of_next_vowel])

def next_consonant(c):
    index_of_c = alphabet.find(c)
    for i in range(index_of_c+1, len(alphabet)):
        if (not is_vowel(alphabet[i])):
            return alphabet[i]
    return 'z'


output_string = ""
for i in range(0, len(word)):
    c = word[i]
    if (is_vowel(c)):
        output_string += c
    else: # c is not vowel, i.e. consonant
        output_string += transform(c)

print(output_string)